### PR TITLE
💄 Move the content in the mobile menu downward

### PR DESCRIPTION
**Pull request by YosefAshenafi**
# Screenshot
fullscreen

### CODE REVIEW

**Review by jontsai — APPROVED** [review]
LFTM

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -79,3 +59,46 @@ export function Header() {
         </header>
     );
 }
+
+export function PrimaryNav({ colorMode, toggleColorMode }: Props) {
```
Comment (gwainor) [inline src/components/Header/index.tsx]: Please move this to its own file in `src/components/Header/PrimaryNav.tsx`

File: src/components/Header/index.tsx [hunk 2]
```diff
@@ -79,3 +59,46 @@ export function Header() {
...
+                <ThemeToggleButton
+                    colorMode={colorMode}
+                    toggleColorMode={toggleColorMode}
+                />
+            </Nav>
+        </div>
+    );
+}
+
+export function MobileNav({ colorMode, toggleColorMode }: Props) {
```
Comment (gwainor) [inline src/components/Header/index.tsx]: Please move this to its own file in `src/components/Header/MobileNav.tsx`

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -79,3 +59,46 @@ export function Header() {
...
+                <Offcanvas.Title className="d-flex align-items-start">
+                    <ThemeToggleButton
+                        colorMode={colorMode}
+                        toggleColorMode={toggleColorMode}
+                        mobileNav
+                    />
+                </Offcanvas.Title>
+            </Offcanvas.Header>
+            <Offcanvas.Body>
+                <span className="d-lg-none d-lg-none d-xl-none fs-4 fw-bold">
```
Comment (gwainor) [inline src/components/Header/index.tsx]: There are two `d-lg-none`. `lg` addition means apply for large screens and up. So there is no need for `d-xl-none`

File: src/components/Header/index.tsx [hunk 4]
```diff
@@ -79,3 +59,46 @@ export function Header() {
...
+    return (
+        <Navbar.Offcanvas id="main-navbar-nav" placement="end">
+            <Offcanvas.Header closeButton>
+                <Offcanvas.Title className="d-flex align-items-start">
+                    <ThemeToggleButton
+                        colorMode={colorMode}
+                        toggleColorMode={toggleColorMode}
+                        mobileNav
+                    />
+                </Offcanvas.Title>
```
Comment (gwainor) [inline src/components/Header/index.tsx]: Why did you move out the title from here? Does it make sense to you that putting title to `body` while there is a `title` section?

What you could do:
- add `flex-column` to `Offcanvas.Title`
- Write `Books` after `ThemeToggleButton` component.

**Review by gwainor — CHANGES_REQUESTED** [review]

File: src/components/Header/index.tsx (hunk 2 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: done

File: src/components/Header/index.tsx (hunk 3 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: done

File: src/components/Header/index.tsx (hunk 4 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: done with Thanks.

File: src/components/Header/index.tsx [hunk 5]
```diff
@@ -1,13 +1,12 @@
 'use client';
 import React from 'react';
-import Link from 'next/link';
-import { Container, Nav, Navbar, Offcanvas } from 'react-bootstrap';
+import { Container, Navbar } from 'react-bootstrap';
 
-import { CATEGORIES, SITE_NAME } from '@/constants';
 import { useColorMode } from '@/hooks';
 
 import { CategoryNav } from './CategoryNav';
```
Comment (gwainor) [inline src/components/Header/index.tsx]: This import is not used anymore?

File: src/components/Header/PrimaryNav.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { SITE_NAME } from '@/constants';
+import { ColorMode } from '@/types';
+import Link from 'next/link';
+import { Nav, Navbar } from 'react-bootstrap';
+import { ThemeToggleButton } from './ThemeToggleButton';
```
Comment (gwainor) [inline src/components/Header/PrimaryNav.tsx]:
```suggestion
import Link from 'next/link';
import { Nav, Navbar } from 'react-bootstrap';
import { SITE_NAME } from '@/constants';
import { ColorMode } from '@/types';
import { ThemeToggleButton } from './ThemeToggleButton';
```

File: src/components/Header/MobileNav.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { ColorMode } from '@/types';
+import { Navbar, Offcanvas } from 'react-bootstrap';
```
Comment (gwainor) [inline src/components/Header/MobileNav.tsx]:
```suggestion
import { Navbar, Offcanvas } from 'react-bootstrap';
import { ColorMode } from '@/types';
```

File: src/components/Header/MobileNav.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+                <Offcanvas.Title className="d-flex flex-column">
+                    <ThemeToggleButton
+                        colorMode={colorMode}
+                        toggleColorMode={toggleColorMode}
+                        mobileNav
+                    />
+                    <span className="d-lg-none fs-4 fw-bold mt-4">Books</span>
+                </Offcanvas.Title>
+            </Offcanvas.Header>
+            <hr />
```
Comment (gwainor) [inline src/components/Header/MobileNav.tsx]: Remove please. You can use border classes of bootstrap.

**Review by gwainor — CHANGES_REQUESTED** [review]

File: src/components/Header/index.tsx (hunk 5 above)
Comment (YosefAshenafi) [reply src/components/Header/index.tsx]: Removed.

File: src/components/Header/PrimaryNav.tsx (hunk 6 above)
Comment (YosefAshenafi) [reply src/components/Header/PrimaryNav.tsx]: done, BTW it is automatically sorted. but to solve your request I add a space between them.

File: src/components/Header/MobileNav.tsx (hunk 7 above)
Comment (YosefAshenafi) [reply src/components/Header/MobileNav.tsx]: done

File: src/components/Header/MobileNav.tsx (hunk 8 above)
Comment (YosefAshenafi) [reply src/components/Header/MobileNav.tsx]: Ok, removed and handled by bootstrap.

**Review by jontsai — APPROVED** [review]
LGTM
